From 51fe0fbf8d49a8d832795b8a12f0c2f83d290aab Mon Sep 17 00:00:00 2001
From: Andreas Blaesius <andi@unlegacy-android.org>
Date: Wed, 5 Jan 2022 20:38:39 +0100
Subject: [PATCH] Use GitHubs issue form for bug reports

Use GitHubs issue form for bug reports.

- modern look
- user don't need to mess with given markdown parts while filling the issue template

Setup config.yml for general questions and problems with the Raspbian distribution packages.
---
 .github/ISSUE_TEMPLATE/bug_report.md  | 34 ----------
 .github/ISSUE_TEMPLATE/bug_report.yml | 91 +++++++++++++++++++++++++++
 .github/ISSUE_TEMPLATE/config.yml     |  9 +++
 3 files changed, 100 insertions(+), 34 deletions(-)
 delete mode 100644 .github/ISSUE_TEMPLATE/bug_report.md
 create mode 100644 .github/ISSUE_TEMPLATE/bug_report.yml
 create mode 100644 .github/ISSUE_TEMPLATE/config.yml

--- a/.github/ISSUE_TEMPLATE/bug_report.md
+++ /dev/null
@@ -1,34 +0,0 @@
----
-name: Bug report
-about: Create a report to help us fix your issue
-
----
-
-**Is this the right place for my bug report?**
-This repository contains the Linux kernel used on the Raspberry Pi. If you believe that the issue you are seeing is kernel-related, this is the right place. If not, we have other repositories for the GPU firmware at [github.com/raspberrypi/firmware](https://github.com/raspberrypi/firmware) and Raspberry Pi userland applications at [github.com/raspberrypi/userland](https://github.com/raspberrypi/userland). If you have problems with the Raspbian distribution packages, report them in the [github.com/RPi-Distro/repo](https://github.com/RPi-Distro/repo). If you simply have a question, then [the Raspberry Pi forums](https://www.raspberrypi.org/forums) are the best place to ask it.
-
-**Describe the bug**
-Add a clear and concise description of what you think the bug is.
-
-**To reproduce**
-List the steps required to reproduce the issue.
-
-**Expected behaviour**
-Add a clear and concise description of what you expected to happen.
-
-**Actual behaviour**
-Add a clear and concise description of what actually happened.
-
-**System**
- Copy and paste the results of the raspinfo command in to this section. Alternatively, copy and paste a pastebin link, or add answers to the following questions:
-
-* Which model of Raspberry Pi? e.g. Pi3B+, PiZeroW
-* Which OS and version (`cat /etc/rpi-issue`)?
-* Which firmware version (`vcgencmd version`)?
-* Which kernel version (`uname -a`)?
-
-**Logs**
-If applicable, add the relevant output from `dmesg` or similar.
-
-**Additional context**
-Add any other relevant context for the problem.
--- /dev/null
+++ b/.github/ISSUE_TEMPLATE/bug_report.yml
@@ -0,0 +1,91 @@
+name: "Bug report"
+description: Create a report to help us fix your issue
+body:
+- type: markdown
+  attributes:
+    value: |
+      **Is this the right place for my bug report?**
+      This repository contains the Linux kernel used on the Raspberry Pi.
+      If you believe that the issue you are seeing is kernel-related, this is the right place.
+      If not, we have other repositories for the GPU firmware at [github.com/raspberrypi/firmware](https://github.com/raspberrypi/firmware) and Raspberry Pi userland applications at [github.com/raspberrypi/userland](https://github.com/raspberrypi/userland).
+      
+      If you have problems with the Raspbian distribution packages, report them in the [github.com/RPi-Distro/repo](https://github.com/RPi-Distro/repo).
+      If you simply have a question, then [the Raspberry Pi forums](https://www.raspberrypi.org/forums) are the best place to ask it.
+
+- type: textarea
+  id: description
+  attributes:
+    label: Describe the bug
+    description: |
+      Add a clear and concise description of what you think the bug is.
+  validations:
+    required: true
+
+- type: textarea
+  id: reproduce
+  attributes:
+    label: Steps to reproduce the behaviour
+    description: |
+      List the steps required to reproduce the issue.
+  validations:
+    required: true
+
+- type: dropdown
+  id: model
+  attributes:
+    label: Device (s)
+    description: On which device you are facing the bug?
+    multiple: true
+    options:
+      - Raspberry Pi Zero
+      - Raspberry Pi Zero W / WH
+      - Raspberry Pi Zero 2 W
+      - Raspberry Pi 1 Mod. A
+      - Raspberry Pi 1 Mod. A+
+      - Raspberry Pi 1 Mod. B
+      - Raspberry Pi 1 Mod. B+
+      - Raspberry Pi 2 Mod. B
+      - Raspberry Pi 2 Mod. B v1.2
+      - Raspberry Pi 3 Mod. A+
+      - Raspberry Pi 3 Mod. B
+      - Raspberry Pi 3 Mod. B+
+      - Raspberry Pi 4 Mod. B
+      - Raspberry Pi 400
+      - Raspberry Pi CM1
+      - Raspberry Pi CM3
+      - Raspberry Pi CM3 Lite
+      - Raspberry Pi CM3+
+      - Raspberry Pi CM3+ Lite
+      - Raspberry Pi CM4
+      - Raspberry Pi CM4 Lite
+      - Other
+  validations:
+    required: true
+
+- type: textarea
+  id: system
+  attributes:
+    label: System
+    description: |
+      Copy and paste the results of the raspinfo command in to this section.
+      Alternatively, copy and paste a pastebin link, or add answers to the following questions:
+      * Which OS and version (`cat /etc/rpi-issue`)?
+      * Which firmware version (`vcgencmd version`)?
+      * Which kernel version (`uname -a`)?
+  validations:
+    required: true
+
+- type: textarea
+  id: logs
+  attributes:
+    label: Logs
+    description: |
+      If applicable, add the relevant output from `dmesg` or similar.
+
+- type: textarea
+  id: additional
+  attributes:
+    label: Additional context
+    description: |
+      Add any other relevant context for the problem.
+
--- /dev/null
+++ b/.github/ISSUE_TEMPLATE/config.yml
@@ -0,0 +1,9 @@
+blank_issues_enabled: false
+contact_links:
+  - name: "⛔ Question"
+    url: https://www.raspberrypi.org/forums
+    about: "Please do not use GitHub for asking questions. If you simply have a question, then the Raspberry Pi forums are the best place to ask it. Thanks in advance for helping us keep the issue tracker clean!"
+  - name: "⛔ Problems with the Raspbian distribution packages"
+    url: https://github.com/RPi-Distro/repo
+    about: "If you have problems with the Raspbian distribution packages, please report them in the github.com/RPi-Distro/repo."
+
